pkg/ip/ip.go

```diff
@@ -804,5 +804,69 @@ func IPToPrefix(ip net.IP) *net.IPNet {
 
 // IsIPv4 returns true if the given IP is an IPv4
 func IsIPv4(ip net.IP) bool {
-	return ip.To4() != nil
+	return ip != nil && ip.To4() != nil
+}
+
+// IsIPv6 returns if netIP is IPv6.
+func IsIPv6(ip net.IP) bool {
+	return ip != nil && ip.To4() == nil
```

Comment:
small nit `ip != nil` is not necessary since `ip.To4()` is checking for that as well

Reply:
For `IPv6` we need the `ip != nil` because if the `net.IP` passed as the argument is nil the function will return true instead of false.